librad/src/identities/payload.rs

```diff
@@ -725,6 +735,28 @@ mod tests {
         )
     }
 
+    #[test]
+    fn remove_ext() {
+        let mut payload = PersonPayload::new(Person {
+            name: "cloudhead".into(),
+        })
+        .with_ext(UpstreamUser {
+            registered_as: "cloudhead".into(),
+        })
+        .unwrap();
+
+        assert_eq!(
+            payload.get_ext::<UpstreamUser>().unwrap().unwrap(),
+            UpstreamUser {
+                registered_as: "cloudhead".into(),
+            }
+        );
+
+        let result = payload.remove_ext::<UpstreamUser>();
+        assert!(result.is_ok());
```

Comment:
```suggestion
        assert_eq!(result, Ok(UpstreamUser { registered_as: "cloudhead".into() }));
```

small suggestion to make sure get out what we put in. If it's a case of `Debug` impls missing we can also use `assert_matches!`

Reply:
Good one! I changed it to this, since it returns `Result<Option<U>, _>`

``` rust
        assert_eq!(
            result.unwrap(),
            Some(UpstreamUser {
                registered_as: "cloudhead".into()
            })
        );
```

Reply:
I have a feeling you might have force pushed the wrong commit :thinking: I don't see the code as updated :sweat_smile:

Reply:
Oops! Here it is: https://github.com/radicle-dev/radicle-link/pull/582/files#diff-db230ec3b5c04fb040752d237b3eaa60cd2725238e5a0c4d679055dbf4db93b5R755-R761